feat-basic authentication

prathmeshjagtap opened the pull request:
Created basic authentication for Login and Signup 

you can not go to cart and Wishlist unless you are logged in

## Files to  Check 
 
Do check basic auth context  , auth reducer ,login form , signUp form , navbar 

- [ ] src/frontend/contexts/auth-context.js
- [ ] src/frontend/pages/login/LoginForm.jsx
- [ ] src/frontend/pages/signup/SignupForm.jsx
- [ ] src/frontend/components/navbar/Navbar.jsx


## What not to check

Have not done too much error Handling 


## To do testing 

Email: test@gmail.com
Password : test@1234

prathmeshjagtap committed:
feat-basic authentication

SukanyaSen530 commented on src/frontend/contexts/auth-context.js:
```diff
@@ -0,0 +1,48 @@
+
+	const authReducer = (state, action) => {
+		switch (action.type) {
+			case authActions.FIRST_NAME:
```
first name, last name, email and password can be stored as an object.

prathmeshjagtap replied:
Yes added

SukanyaSen530 reviewed (approved):
Everything else look's great :)

vijaytembugade commented on src/index.js:
```diff
@@ -4,20 +4,26 @@ import "./index.css";
-				</FilterProvider>
-			</DataProvider>
-		</BrowserRouter>
+		<AuthProvider>
```
BrowerRouter > AuthProvider, Any particular reason?

prathmeshjagtap replied:
No reason as such.

vijaytembugade reviewed (approved):
Good work prathamesh.

prathmeshjagtap committed:
feat-added user object

Added user object for all the user details.
added logout functionality,
added authReducer in separate folder
